tests/unittests/AddressBook/ModifyContactsTests.mm

```diff
@@ -0,0 +1,241 @@
+                                 (__bridge NSString*)kABOtherLabel
+                              ],
+                              @[
+                                 @"0000000000",
+                                 @"1111111111",
+                                 @"2222222222",
+                                 @"3333333333",
+                                 @"4444444444",
+                                 @"5555555555",
+                                 @"6666666666",
+                                 @"7777777777",
+                                 @"8888888888",
+                                 @"9999999999"
+                              ]),
+        ::testing::make_tuple(kABPersonURLProperty,
+                              @[ (__bridge NSString*)kABPersonHomePageLabel, (__bridge NSString*)kABPersonHomePageLabel ],
+                              @[ @"bing.com", @"microsoft.com" ]),
+        ::testing::make_tuple(kABPersonEmailProperty,
+                              @[ (__bridge NSString*)kABHomeLabel, (__bridge NSString*)kABWorkLabel, (__bridge NSString*)kABOtherLabel ],
+                              @[ @"a@ms.com", @"b@ms.com", @"c@ms.com" ]),
+        ::testing::make_tuple(kABPersonRelatedNamesProperty,
+                              @[
+                                 (__bridge NSString*)kABOtherLabel,
+                                 (__bridge NSString*)kABPersonSpouseLabel,
+                                 (__bridge NSString*)kABPersonPartnerLabel,
+                                 (__bridge NSString*)kABPersonSiblingLabel,
+                                 (__bridge NSString*)kABPersonParentLabel,
+                                 (__bridge NSString*)kABPersonChildLabel
+                              ],
+                              @[ @"Random Name", @"Wife's Name", @"Buzz", @"Link", @"Pops", @"Sun Ray" ])
+
+            ));
+
+TEST(AddressBookModify, DateMultiValues) {
+    ABRecordRef person = ABPersonCreate();
+
+    ABMutableMultiValueRef multiValue = ABMultiValueCreateMutable(kABDateTimePropertyType);
+
+    NSCalendar* calendar = [NSCalendar currentCalendar];
+    NSDateComponents* dateComponents1 = [[NSDateComponents alloc] init];
+    dateComponents1.day = 4;
+    dateComponents1.month = 5;
+    dateComponents1.year = 1975;
+    NSDate* date1 = [calendar dateFromComponents:dateComponents1];
+    NSDateComponents* dateComponents2 = [[NSDateComponents alloc] init];
+    dateComponents2.day = 4;
+    dateComponents2.month = 6;
+    dateComponents2.year = 1935;
+    NSDate* date2 = [calendar dateFromComponents:dateComponents2];
+    [dateComponents1 release];
+    [dateComponents2 release];
+
+    ABMultiValueAddValueAndLabel(multiValue, (__bridge CFDateRef)date1, (__bridge CFStringRef) @"hi", NULL);
+    ABMultiValueAddValueAndLabel(multiValue, (__bridge CFDateRef)date2, (__bridge CFStringRef) @"yo", NULL);
+
+    ASSERT_TRUE(ABRecordSetValue(person, kABPersonDateProperty, multiValue, NULL));
```

Comment:
Will add cases for this, but errors would occur if a String property's length was longer than the allowed Windows length, or if an unknown property was attempted to be set

---
In reply to: [77093837](https://github.com/Microsoft/WinObjC/pull/935#discussion_r77093837) [](ancestors = 77093837,76670902)